vmdb/lib/workers/mixins/event_catcher_openstack_mixin.rb

```diff
@@ -0,0 +1,74 @@
+module EventCatcherOpenstackMixin
+  # seems like most of this class could be boilerplate when compared against EventCatcherRhevm
+  def event_monitor_handle
+    require 'openstack/openstack_event_monitor'
+    unless @event_monitor_handle
+      options = {}
+      options[:hostname] = @ems.ipaddress
+      options[:port]     = self.worker_settings[:amqp_port]
+      if @ems.has_authentication_type? :amqp
+        options[:username] = @ems.authentication_userid(:amqp)
+        options[:password] = @ems.authentication_password(:amqp)
+      end
+      options[:topics] = self.worker_settings[:topics]
+      options[:duration] = self.worker_settings[:duration]
+      options[:capacity] = self.worker_settings[:capacity]
+      @event_monitor_handle = OpenstackEventMonitor.new(options)
+    end
+    @event_monitor_handle
+  end
+
+  def reset_event_monitor_handle
+    @event_monitor_handle = nil
+  end
+
+  def stop_event_monitor
+    begin
+      @event_monitor_handle.stop unless @event_monitor_handle.nil?
+    rescue Exception => err
+      $log.warn("#{self.log_prefix} Event Monitor Stop errored because [#{err.message}]")
+      $log.warn("#{self.log_prefix} Error details: [#{err.details}]")
+      $log.log_backtrace(err)
+    ensure
+      reset_event_monitor_handle
+    end
+  end
+
+  def monitor_events
+    begin
+      event_monitor_handle.start
+      event_monitor_handle.each_batch do |events|
+        $log.debug("#{self.log_prefix} Received events #{events.collect {|e| e.payload["event_type"]}}") if $log.debug?
```

Comment:
So, I really don't know what topics/queues the infrastructure side is using here.  All I know is that when I logged into an undercloud and listed out the exchanges and queues in the AMQP service, there was nothing like I saw when I did the cloud event integration.

I think you may have to dig through the list of queues, or talk to someone who knows more about the way notifications are working in the undercloud.

The bottom line is I don't think the same list of topics and queues that work for the overcloud will work for the undercloud.